src/Core/PostgreSQL/Utils.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <pqxx/pqxx> // Y_IGNORE
+#include <Core/Types.h>
+#include "Connection.h"
+
+namespace pqxx
+{
+    using ReadTransaction = pqxx::read_transaction;
+    using ReplicationTransaction = pqxx::transaction<isolation_level::repeatable_read, write_policy::read_only>;
+}
+
+
+namespace postgres
+{
+
+ConnectionInfo formatConnectionString(String dbname, String host, UInt16 port, String user, String password);
+
+Connection createReplicationConnection(const ConnectionInfo & connection_info);
+
+template <typename T>
+class Transaction
+{
+public:
+    Transaction(pqxx::connection & connection) : transaction(connection) {}
+
+    ~Transaction() { transaction.commit(); }
```

Comment:
It is quite unlikely, but it is not possible to guarantee otherwise - anything can happen (connection loss, something else - the library rethrows such errors), but why is it not ok? Basically all operations, which create transaction object are under try - catch in the code of replicaiton handler and replication consumer.

Reply:
It's usually dangerous to throw from destructor, possibly we should add try-catch in destructor to ignore exception or call `commit()` explicitly (not from destructor) if we want to handle exception somehow. See https://akrzemi1.wordpress.com/2011/09/21/destructors-that-throw/ for details